test/e2e-test/rule-test/util/test-rule-functions.js

```diff
@@ -0,0 +1,145 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+var testUtils = require('./utils');
+var fhirModule = require('fhir');
+var fhir = new fhirModule.Fhir();
+const _ = require('lodash');
+const path = require('path');
+const { v4: uuidv4 } = require('uuid');
+const fs = require('fs');
+const fsExtra = require('fs-extra');
+const systemExec = require('child_process').execSync;
+
+const MAX_REVEAL_DEPTH = 100;
+const NON_COMPARE_PROPERTY = new Set([
+    'resourceType', 'type', 'fullUrl', 'id', 'method', 'url', 'reference', 'system', 'code', 'display',
+    'gender', 'use', 'preferred', 'status'
```

Comment:
How are these properties chosen?

Reply:
They are chosen by practice. Run the tests once and once, select the property which breaks the correction, then paste it inside the set. Of course the semantics will be rechecked before adding.

Reply:
Could you please give more details why these properties will break the comparison?

Reply:
For example, `use="work"`, the property value `"work"` can't be found in the origin data.

Reply:
Got it. Thanks!